src/mondeath.cpp

```diff
@@ -97,6 +97,37 @@ void mdeath::boomer(monster *z)
     }
 }
 
+void mdeath::boomer_glow(monster *z)
+{
+    player *ch = dynamic_cast<player*>( z->get_killer() );
+    std::string explode = string_format(_("a %s explode!"), z->name().c_str());
+    sounds::sound(z->pos(), 24, explode);
+
+
+    for (int i = -1; i <= 1; i++) {
+        for (int j = -1; j <= 1; j++) {
+            tripoint dest( z->posx() + i, z->posy() + j, z->posz() );
+            g->m.bash(dest , 10 );
+            g->m.add_field(dest , fd_bile, 1, 0);
+            int mondex = g->mon_at(dest);
+            if (mondex != -1) {
+                g->zombie(mondex).stumble(false);
+                g->zombie(mondex).moves -= 250;
+                }
+        }
+    }
+    if (rl_dist( z->pos(), g->u.pos() ) == 1) {
+        g->u.add_env_effect("boomered", bp_eyes, 5, 25);
+        for (int i = 0; i < rng(2,4); i++){
+                body_part bp = random_body_part();
+                g->u.add_env_effect("glowing", bp, 4, 40);
+                if (ch->has_effect("glowing")){
```

Comment:
I'm pretty sure (didn't test though) that this line could be made to crash.
Let's say you interpose a giant boomer between self and a hulk. Hulk will smash the boomer killing it (and becoming the killer). So `ch` would be null here.

You don't even need to cast anything here, though. `Creature` class also has the `has_effect` function. In general, you shouldn't cast classes unless you have no other choice.